### ISSUE TITLE
H53 Using the body of the object element is not practically supported

### ISSUE THREAD

**Issue by mraccess77** [w3c/wcag#1138]
In my experience users of assistive technology will never encounter the inner text of an object element unless the object is not available and the text is displayed as fallback content by the browser.

I recommend this technique either be removed or modified to communicate that the technique does not provide an alternative unless it's fallback when the object can't be displayed -- and then it's a fallback for all users.

https://www.w3.org/WAI/WCAG21/Techniques/html/H53

**Comment by alastc** [comment]
So the typical questions we have when trying to deprecate a technique are:
- Is it required for any of the SC, e.g. does removing it undermine an SC?
- Are people still using objects for this?

Also, can anyone draft an update as @mraccess77 suggested?

**Comment by WilcoFiers** [comment]
This recently came up in ACT too. Totally agree with Jon on this. The content of the `object` is a fallback, it isn't used in generating an accessible name, it never has been. The same is true for `iframe` and `canvas`.

To answer the questions:

1. Removing the technique does not impact any SC, except that it makes it clear that this, which doesn't work, and isn't supposed to work, shouldn't be used.
2. If anyone is using objects like this, they have an accessibility problem because of it.

**Comment by yecril71pl** [comment]
An object is supposed to display a some content.  If the content is inaccessible, it is a problem of the content, not of the object.

**Comment by kantoche** [comment]
I agree with @WilcoFiers and @mraccess77 and I am in favor of deprecating the H53 technique.

Two additional comments:

- The same attention should be given to the H46 technique: [Using `noembed` with `embed`](https://www.w3.org/WAI/WCAG21/Techniques/html/H46), even if the current `embed` element no longer accepts a fallback content anyway.
- @WilcoFiers , when you say: 

> The same is true for iframe and canvas

I disagree with you concerning the canvas element, because the fallback content of a canvas element doesn't work the same way as that of an object element ; as the spec states:

> In non-visual media, and in visual media if [scripting is disabled](https://html.spec.whatwg.org/multipage/webappapis.html#concept-n-noscript) for the [canvas](https://html.spec.whatwg.org/multipage/canvas.html#the-canvas-element) element or if support for [canvas](https://html.spec.whatwg.org/multipage/canvas.html#the-canvas-element) elements has been disabled, the [canvas](https://html.spec.whatwg.org/multipage/canvas.html#the-canvas-element) element [represents](https://html.spec.whatwg.org/multipage/dom.html#represents) its [fallback content](https://html.spec.whatwg.org/multipage/dom.html#fallback-content) instead. 

(from https://html.spec.whatwg.org/multipage/canvas.html#the-canvas-element)

So, even if the fallback content is a less good alternative than using an img role and an attribute providing the accessible name, it is still a possibility.
You seem to say that ACT also discusses this topic, but I haven't found an issue specifically related to it: would you have a link to suggest?

**Comment by yecril71pl** [comment]
1. The fallback content may need alternative text too that it should provide itself.
2. The replacement content (object data) may provide alternative text itself (e.g. a Description attribute in image metadata).  If it does, the alternative text declared on the `object` itself should not be used.  This will lead to situations where the coder knows that the alternative text will be replaced along with the object and will fill it with ~lorem ipsum~ to hush the test.

**Comment by WilcoFiers** [comment]
Iframes, object, and canvas all use its content as fallback. Iframes for when frames are turned off, objects for when the src doesn't load, and canvas for when the script is turned off.

In all cases, the content isn't included in the accessible name of the elements AFAIK.

**Comment by yecril71pl** [comment]
> Over the past few years, more and more screen readers have begun to support `aria-labelledby` for the `<object>` element. ☞[`object` with `aria-labelledby` attribute - Screen reader compatibility](https://www.powermapper.com/tests/screen-readers/media/object-aria-labelledby/)
> 
> Therefore, even if H53 is not deprecated, it can be adapted to modern situations by, for example, modifying [Example 3](https://www.w3.org/WAI/WCAG21/Techniques/html/H53#example-3-the-image-object-has-content-that-provides-a-brief-description-of-the-function-of-the-image) as follows
> 
> ```
> <object data="companylogo.gif" type="image/gif" aria-labelledby="image-label">
>   <p id="image-label">Company Name</p>
> </object>
> ```

An embedded object does not render its fallback content unless fallback is needed.  This may occur if the content is unavailable or unsupported.  It does not occur when the content is inaccessible to the user. 
 Accordingly, it is inconsistent to require assistive technology to use fallback content in any way.  The only element of the `OBJECT` element you can reasonably expect to be present is the `PARAM` element&mdash;but it would also be abusive to use it for accessibility.

**Comment by innoticFR** [comment]
I agree with you on this subject, and I am surprised it hasn't already been removed from 1.1.1. Is there any plan for this soon?

**Comment by patrickhlauke** [comment]
@innoticFR well last activity on this was 4 years ago, and nobody's gone ahead to actually make a PR/proposal ... "PRs welcome", otherwise it comes down to "when the handful of folks working on the WCAG 2.x backlog TF get to it among the 400+ open issues that are currently filed" 🥲

**Comment by innoticFR** [comment]
Looking closely at the description, the important part is:

> Authors can only rely on this technique to satisfy the success criterion if they are not relying on the direct accessibility of the media's technology in their conformance claim, and reasonably expect users will be able to access the fallback.

This means, for instance, that an SVG with an appropriate <title> tag can use this technique, but the author must ensure that the fallback content itself is accessible

Maybe changing the first paragraph could be a first step:

> The objective of this technique is to provide an additional text alternative for the content rendered using the object element, in order to handle cases where the technology is not supported, if not relied-upon.
